contrib/go-chi/chi/chi.go

```diff
@@ -0,0 +1,54 @@
+// Package chi provides tracing functions for tracing the go-chi/chi package (https://github.com/go-chi/chi).
+package chi // import "gopkg.in/DataDog/dd-trace-go.v1/contrib/go-chi/chi"
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/go-chi/chi"
+	"github.com/go-chi/chi/middleware"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// Middleware returns middleware that will trace incoming requests.
+func Middleware(service string) func(next http.Handler) http.Handler {
```

Comment:
This is a mistake we regrettably made with the `gin-gonic/gin` integration, which makes it hard to add new options to this middleware now because it would result in changing the arguments to this function, which would be a breaking change 😕 . Do you think we could use a set of functional options here like we do in other integrations (such as `gorilla/mux`) ? Then this would look something like:
```go
 // Middleware returns middleware that will trace incoming requests.
func Middleware(opts ...Option) func(next http.Handler) http.Handler {
    	cfg := new(config)
	defaults(cfg)
	for _, fn := range opts {
		fn(cfg)
	}
        return func(next http.Handler) http.Handler {
            // ... use service name from cfg
        }
}
```
We are adding more features these days and this will come in handy. This way we can keep adding them without breaking the API.

Reply:
For inspiration: https://github.com/DataDog/dd-trace-go/blob/v1/contrib/gorilla/mux/mux.go#L70-L74

Reply:
Added functional options. Check out option.go